pkg/storage/transactions_test.go

```diff
@@ -37,47 +40,95 @@ func (suite *StoreTestSuite) TestWithTransaction() {
 	})
 
 	suite.Run("Errors will rollback a transaction", func() {
-		newTRB, err := sqlutils.WithTransaction[models.TRBRequest](suite.store, func(tx *sqlx.Tx) (*models.TRBRequest, error) {
+		var createId uuid.UUID
+		err := sqlutils.WithTransaction(ctx, suite.store, func(tx *sqlx.Tx) error {
```

Comment:
Perhaps this should test WithTransactionRet still? I think the original intent was to verify the zero state of the return value when rolled back.

Reply:
We should chat about this! The return value shouldn't matter when there is an error assuming correct error handling. We have to `return result, txErr` due to the nature of the deferred func though vs. something like:

```go
if txErr != nil {
    return nil, txErr
}

return result, nil
```

Reply:
Maybe we should talk about if transaction helpers should even return a value? maybe it should only ever be `err`?

Reply:
Hm - ok I see what you are saying. Will push up something momentarily! Thanks!

Reply:
Thanks Sam! Looks good!